index.js

```diff
@@ -102,6 +102,22 @@ var publicAgent = {
       return this;
     }
 
+    common.utils.getHostname(function(err, hostname) {
+      if (err) {
+        logger.info('Unable to retrieve GCE hostname.', err);
+      } else {
+        config.hostname = hostname;
+      }
+    });
+
+    common.utils.getInstanceId(function(err, instanceId) {
+      if (err) {
+        logger.info('Unable to retrieve GCE instance id.', err);
+      } else {
+        config.instanceId = instanceId;
+      }
+    });
+
```

Comment:
1. This is not an actionable item, but rather an observation. Any traces captured before these async functions complete will not have the instance and hostname fields attached to them. I do think it is worth capturing traces even if these operations haven't completed.

2. Upgrade the logger message to `.warn` instead of `.info`. If we are running on GCP and can't get this data, this is a problem; albeit not a critical one.

3. When not running on GCP, neither a `warn` or `info` is warranted.